### PR TITLE
More examples

### PR DESCRIPTION
**Pull request by anthophobiac**
- Iterate over table rows
- Mock date
- Wait for file download

### CODE REVIEW

**Comment by VasilyStrelyaev** [comment]
Please rebase onto `upstream/master` to resolve the conflicts.

File: examples/iterate-over-table-rows/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Iterate Over Table Rows
+
+**Test Code**: [index.js](index.js)
+
+This example shows how to iterate over table rows.
```
Comment (dirk-pieterse) [inline examples/iterate-over-table-rows/README.md]: Why iterate 'over'?

File: examples/iterate-over-table-rows/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+The [example page](index.html) contains a table over which the test code iterates. Select the table and get the total number of rows in it:
```
Comment (dirk-pieterse) [inline examples/iterate-over-table-rows/README.md]: same here

File: examples/iterate-over-table-rows/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+```js
+const table        = Selector('table');
+const dataRows     = table.find('tbody > tr');
+const dataRowCount = await dataRows.count;
+```
+
+When `dataRowCount` is available, the test can iterate over the rows. In the example, the iteration is done in a `for` loop.
+
+In each row, the test checks the cell where the index is equal to one and whether it contains the text `Smith`. If the following row is even, the test checks if this row has a checkbox in a checked state.
```
Comment (dirk-pieterse) [inline examples/iterate-over-table-rows/README.md]:
```suggestion
In each row, the test checks which cell's index is equal to one and whether it contains the text `Smith`. If the following row is even, the test checks if this row has a checkbox in a checked state.
```

Comment (dirk-pieterse) [reply examples/iterate-over-table-rows/README.md]: row is even? Explain

File: examples/mock-date/README.md [hunk 4]
```diff
@@ -1 +1,9 @@
-#How to mock data
+# How To Mock Date
+
+**Test Code**: [index.js](index.js)
+
+[`mockdate`](https://www.npmjs.com/package/mockdate) is an npm module that allows you to substitute the current date.
+
+Use the `clientScripts` function to inject the `mockdate` module and add the code that uses this module to set the browser date value with `MockDate.set`, passing a string with the date as its argument.
```
Comment (dirk-pieterse) [inline examples/mock-date/README.md]: This sentence does not make sense. What are you trying to say here?

File: examples/mock-date/README.md [hunk 5]
```diff
@@ -1 +1,9 @@
...
+
+The client code returns the date and the test code verifies that the date equals the one specified in the `clientScript` function.
```
Comment (dirk-pieterse) [inline examples/mock-date/README.md]:
```suggestion
The client code returns the date and the test code verifies that the date is equal to the date in the `clientScript` function.
```

File: examples/wait-for-file-download/README.md [hunk 6]
```diff
@@ -1 +1,9 @@
-# Wait for file download
+# Wait For File Download
+
+**Test Code**: [test.js](test.js)
+
+This example shows how to set a test to wait for the file download.
```
Comment (dirk-pieterse) [inline examples/wait-for-file-download/README.md]:
```suggestion
This example shows how to specify that a test should to wait for the file to download.
```

File: examples/wait-for-file-download/README.md [hunk 7]
```diff
@@ -1 +1,9 @@
...
+
+Before the test begins, it obtains the `downloadedFilePath`. If the file already exists on the path, the test removes the file to save the download without warnings or errors.
```
Comment (dirk-pieterse) [inline examples/wait-for-file-download/README.md]:
```suggestion
Before the test begins, it obtains the `downloadedFilePath`. If the file already exists, the test removes the file and saves the downloaded file without warnings or errors.
```

File: examples/wait-for-file-download/README.md [hunk 8]
```diff
@@ -1 +1,9 @@
...
+
+In the example, the test runs on Google Chrome browser. The asynchronous `waitForDownload` function runs a `for` loop which checks if the file exists on the specified download path. It runs the body of the loop several times, with a timeout between tries, until the file is downloaded. If the function does not locate the file after the tries, it returns `false`.
```
Comment (dirk-pieterse) [inline examples/wait-for-file-download/README.md]: "exists on the specified download path" Why 'on' the path?

Comment (dirk-pieterse) [reply examples/wait-for-file-download/README.md]:
```suggestion
In the example, the test runs on Google Chrome browser. The asynchronous `waitForDownload` function runs a `for` loop which checks if the file exists in the specified download path. The body of the loop is run several times with a timeout between attempts until the file is downloaded. If the function cannot locate the file, it returns `false`.
```

File: examples/iterate-over-table-rows/README.md (hunk 3 above)
Comment (anthophobiac) [reply examples/iterate-over-table-rows/README.md]: @dirk-pieterse table row the index of which can be divided by two is called even, the rest are called odd. Example of usage can be found [here](https://www.w3.org/Style/Examples/007/evenodd.en.html)

File: examples/iterate-over-table-rows/README.md (hunk 1 above)
Comment (anthophobiac) [reply examples/iterate-over-table-rows/README.md]: @dirk-pieterse `iterate over` is a much more common way to phrase this, especially when it comes to programming concepts. There is some info on it [here](https://stackoverflow.com/questions/2715388/is-it-iterate-through-or-iterate-over-something).

File: examples/iterate-over-table-rows/README.md (hunk 2 above)
Comment (anthophobiac) [reply examples/iterate-over-table-rows/README.md]: Same as [here](https://github.com/DevExpress/testcafe-examples/pull/11#discussion_r392165191)

File: examples/mock-date/README.md (hunk 4 above)
Comment (anthophobiac) [reply examples/mock-date/README.md]: @dirk-pieterse the function injects (introduces) the npm module we've mentioned before and then we add the code that alters the browser date value. We use the `mockdate` module to do that. The function `MockDate.set` takes a new date as an argument. This allows substituting the current date with another date.

File: examples/iterate-over-table-rows/README.md (hunk 1 above)
Comment (dirk-pieterse) [reply examples/iterate-over-table-rows/README.md]: Ok. I'm familiar with programming concepts, such as 'to iterate something', but I've not come across 'iterate over' before.

File: examples/iterate-over-table-rows/README.md (hunk 3 above)
Comment (dirk-pieterse) [reply examples/iterate-over-table-rows/README.md]: Ok

File: examples/mock-date/README.md (hunk 4 above)
Comment (dirk-pieterse) [reply examples/mock-date/README.md]: 'passing a string with the date as its argument' passing it to what?

**Pull request merged**